Angular/src/app/project-info/resources-table-for-project/resources-internal-table/resources-internal-table.component.html

```diff
@@ -0,0 +1,32 @@
+<div class="mat-elevation-z8">
+  <table mat-table [dataSource]='dataSource'>
+
+    
+    <ng-container matColumnDef="id">
+      <th mat-header-cell *matHeaderCellDef> ID  </th>
+      <td mat-cell *matCellDef="let element; let i = index"> {{i+1}} </td>
+    </ng-container>
+
+   
+    <ng-container matColumnDef="name">
+      <th mat-header-cell *matHeaderCellDef> Назва </th>
+      <td mat-cell *matCellDef="let element"> {{element.name}} </td>
+    </ng-container>
+
+
+    <ng-container matColumnDef="date">
+      <th mat-header-cell *matHeaderCellDef> Дата створення </th>
+      <td mat-cell *matCellDef="let element"> {{element.createdDateUtc | customDate}} </td>
+    </ng-container>
+
+
+    <ng-container matColumnDef="description">
+      <th mat-header-cell *matHeaderCellDef> Опис </th>
+      <td mat-cell *matCellDef="let element"> {{element.description}} </td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="displayedColumns; sticky: true" ></tr>
+    <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
+  </table>  
+  <mat-paginator [length]="lengthDataArrForDataSource" [pageSize]="pageSize" [pageSizeOptions]="[5, 10, 20]" showFirstLastButtons></mat-paginator>
```

Comment:
remind me, what is the difference between this and general-resources-inner-table.component.html?

Reply:
they use different data acquisition structures. one takes all the data at once and then uses it and the second takes the data through pagination

Reply:
different way of receiving, processing and presenting data